### DISCUSSION TITLE
The Santoker r500master is not supported by Artisan？？？

### DISCUSSION THREAD

**Discussion by lincenxie** [artisan-roaster-scope/artisan#1019] [Q&A]
Hi, I’m looking for help, connecting my Santoker r500master roaster's wifi  Can’t connect to artisan-win-2.8.0 but I keep getting "MODBUS communication error".The Santoker r500master is not supported by Artisan？？？

**Comment by bigl00z3** [comment]
Have you checked this page [https://artisan-scope.org/machines/santoker/](https://artisan-scope.org/machines/santoker/) ?

and also in Artisan  `Config>>Machine>>Santoker `:
<img width="482" alt="image" src="https://user-images.githubusercontent.com/8886922/204159346-d7d35e95-2368-43fa-88ee-6fba2a597776.png">

When looking at your [Roaster's manufacturer web page ](https://coffeeroastco.com/collections/santoker/products/santoker-r500-coffee-roaster-700g-batch) I see: "_2-in-1 Mobile Phone Bluetooth and Computer USB work with Artisan Software for data logging_" Don't they provide instructions in their user manual?

Nico

**Reply by lincenxie** [reply]
Thanks a lot Nico
[https://wintoproasters.com/ws-500-master-automatic-coffee-roaster/](url)
Machine: Bluetooth and PC USB are not supported! 
Connection: TCP over network Wi-Fi
How can I set it up Santoker (over network) Roasting
Hostname/IP address: 10.10.100.254
Port: 20001
[
![01](https://user-images.githubusercontent.com/119077677/204903160-e0327593-098f-45cb-84be-f4ccac47ab98.png)
](url)[
![02](https://user-images.githubusercontent.com/119077677/204902693-8fe33d06-03ed-496d-bf34-57ea82d5c8bd.png)
](url)[
![03](https://user-images.githubusercontent.com/119077677/204902749-33d5e1fe-4444-4cef-83c4-a0efae92bdde.png)
](url)[
![04](https://user-images.githubusercontent.com/119077677/204902795-30737e3e-21fe-40bc-92c6-69d97d6c7b9e.png)
](url)

**Reply by bigl00z3** [reply]
If I look at the link you posted, I see a mention of Siemens (probably S7) not Modbus:

● Wifi-connection to Mobile phone ( or Pad) APP for creating and repeating roasting profiles.
● App Software operation or 7” Siemens touch screen control panel operation.

But maybe I'm wrong.

**Reply by lincenxie** [reply]
Thanks a lot Nico
I tried all possible options with no success

**Comment by MAKOMO** [comment]
Your machine seems to be a newer model which might use the same communication pattern like the Santoker Sample Roaster Q5. The app available from the manufacture for both those machines seem to be at least similar. The protocol those machines talk is unknown to us and exactly those two machines do not have an "Artisan" tag on their page. The corresponding manual model product page holds is tag "Bluetooth + USB data logger:  goroasters, Roastmaster, Artisan". I fear only the manufacture can clarify this.

**Reply by lincenxie** [reply]
Thank you very much MAKOMO
As you said the machine uses the same communication mode as the Santoker Sample Roaster Q5
Currently only Cropster Roasting Intelligence Beta can establish connection and control, but its function is not very friendly
[
![01](https://user-images.githubusercontent.com/119077677/204991066-34a48ffd-5195-4824-a604-001fd130c986.png)
](url)
[https://help.cropster.com/en/knowledge/santoker-over-network-roasting-intelligence-ri-setup](url)
i still love Artisan
Thank you again

**Reply by MAKOMO** [reply]
I implemented something. Could you test it on your machine?

The current [continuous build](https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous) contains a Santoker R Series setup which might work.

**Reply by lincenxie** [reply]
perfect connection
thank you very much MAKOMO

**Reply by MAKOMO** [reply]
Everything works? Control sliders and buttons on Artisan are working too? Also the drum speed slider? I am not sure this machine offers variable drum speed. Is something missing? Any suggestion on how to improve this initial implementation?

**Reply by MAKOMO** [reply]
What about those extra buttons "Machine ON/OFF", "Burner ON/OFF", "Cooling ON/OFF", "Stirrer ON/OFF" and "Warm UP ON/OFF"? Do the work and are the labeled correctly? What does "Warm UP" do?

**Reply by lincenxie** [reply]
Hello MAKOMO
extra buttons can work normally "Machine ON/OFF", "Burner ON/OFF", "Cooling ON/OFF", "Stirrer ON/OFF"The machine can receive instructions The main board of the machine only provides a few high-level signals and no control module is installed at the back end, which requires DIY installation。
Regarding "Warm UP ON/OFF", you need to set the required warm-up temperature on the device. When the preset temperature is reached, the heating will be automatically stopped, and the heating will be turned on automatically after the temperature drops 20 degrees.

fan slider and Drum slider Useless Drums and fans are only manually controlled, no variable speeds are offered

The Artisan power slider can only receive and not output command control, this needs to be adjusted slightly

Additional buttons are still missing (end) This button is to automatically open the bean outlet door to start cooling and stirring at the same time.

**Comment by lincenxie** [comment]
Hello MAKOMO
extra buttons can work normally       "Machine ON/OFF", "Burner ON/OFF", "Cooling ON/OFF", "Stirrer ON/OFF"The machine can receive instructions The main board of the machine only provides a few high-level signals and no control module is installed at the back end, which requires DIY installation。
Regarding "Warm UP ON/OFF", you need to set the required warm-up temperature on the device. When the preset temperature is reached, the heating will be automatically stopped, and the heating will be turned on automatically after the temperature drops  20 degrees.

> fan slider and Drum slider Useless  Drums and fans are only manually controlled, no variable speeds are offered

> The Artisan power slider can only receive and not output command control, this needs to be adjusted slightly

>Additional buttons are still missing (end) This button is to automatically open the bean outlet door to start cooling and stirring at the same time.

**Reply by MAKOMO** [reply]
Ok, so I remove the slider control, but keep the event reading (quantifiers) for the power. I also hide the air/drum LCDs and the corresponding sliders as they do not receive any data (completely manual). I keep those extra buttons, like ON/OFF,...  for now. The END signal is send on DROP to the machine, however, I had a typo in that command (it was sending SCs instead). I will fix this. The other signals are send to the machine on pressing the Artisan buttons, like CHARGE, DRY, FCs and SCs too. Not sure what those actions do on the machine.

I updated the continuous build accordingly (new builds should be available in latest 30min under the same link). To test you need to download and install the app again and run the Santoker R Series machine setup again.

**Reply by MAKOMO** [reply]
Could you please test the current build again and report back? https://github.com/artisan-roaster-scope/artisan/releases/tag/continuous

Thanks!

**Reply by MAKOMO** [reply]
After installation you need to run the machine setup again under menu Config >> Santoker >> R Series

**Reply by lincenxie** [reply]
Hello MAKOMO
May not be able to work recently due to COVID-19, I will get back to you with test results ASAP

**Reply by MAKOMO** [reply]
Uh! Don't worry and get better soon. We will release what we have next week and rework if needed afterwards. A good recovery to you.

**Reply by lincenxie** [reply]
Hello MAKOMO
Install 2.8.2 works perfectly, thanks a lot for your help
I still have a question and need your help, how to make the machine run automatically after the curve design is completed

**Reply by MAKOMO** [reply]
Thanks for testing again and for this confirmation. Your question is on automating a roast? You can either replay events, develop a set of alarm rules to automate your burner control or use the Artisan software PID to duplicate a background profile thus on a drum roaster this is not as effective as on a (small) air roaster. For the PID control see https://artisan-roasterscope.blogspot.com/2016/11/pid-control.html
